### ISSUE TITLE
AIR 33.1.1.50 - Android - Bringing app from background make it lose 3d context

### ISSUE THREAD

**Issue by alexXey62353** [Gamua/Adobe-Runtime-Support#256]
AIR 33.1.1.50
Android
Starling framework

Putting the app to background and then bringing it back, makes it lose it's 3d context.

**Known Workarounds**

We reverted to AIR 33.0.2.338 and the problem disappeared.

**Comment by hardcoremore** [comment]
Can you also give a little more details. Which version of Android, what device etc.

**Comment by alexXey62353** [comment]
From my own experience:
Samsung A50 (Android 9)
Nox player sim (Android 4.4.2)

But, we actually have an event in prod to monitor Lost Context and it spiked through the roof with AIR 33.1.1.50 and 33.1.1.63.

**Comment by neuronix** [comment]
Note as well just when swiping the bottom OS menu up on Huawei devices as well. First time you swipe the menu up, context loss for no reeason.

**Comment by ajwfrost** [comment]
Yes we've reproduced this now and are investigating. Seems to be related to some changes we made to try to work around another issue with API 28 changes..

Thanks

**Comment by hardcoremore** [comment]
Hey Andrew,

Just a small followup. In latest release notes there is:

>  Reverting fix for AIR-168 to avoid context recreating when bringing app to foreground (Gamua-256)

What was fix AIR-168 that was reverted?

Thanks

**Comment by ajwfrost** [comment]
@hardcoremore that was #67  (which affects fewer people,it's a similar issue/cause though)

**Comment by ventr1x** [comment]
Is there any movement on this issue?
We are currently having the context loss issue on any type of app suspension since AIR33.1.x on Android (any version, any device we've tested).
We also experience an extreme lag/delay on every context loss, making for horrible (basically unusable for anyone switching between apps often) user experience.

Video of the issue:
https://drive.google.com/file/d/1hu2985Qk9L7oU701u18bZyvUwDqo7Dsz/view

We are also desperately trying to contact any kind of Harman support for days now. As an Enterprise tier licensee I would expect some more help on issues introduced by Harman and pushed to production...

**Comment by ajwfrost** [comment]
@ventr1x  can I just check - this is something we had down as an issue introduced in 33.1.1.50 and fixed in 33.1.1.98. Can you confirm that you're using 33.1.1.98? It may be a different root cause perhaps, but if you're seeing the context loss notification from the runtime then we can look into this further.

It would be very useful if you're able to build your application using 33.1.1.98 and the "apk-debug" setting, and then send this to us as we can then start hooking into the C++ side to see what's happening within EGL.. below is a link for uploading a file in case that's feasible?
https://transfer.harman.com/requests/Xlt6cyxR2bK5YyUD2T6MfG

thanks

**Comment by ventr1x** [comment]
My latest sdk download (for MacOS) was end of June which still gave me x.63. I guess it would be useful to start adding dates to your release notes. It's not easy to keep track this way.

I just tried redownloading multiple times and it redirected me a few times to a .txt file stating "invalid download link".

I'll test the newest version now and update you after a short test

**Comment by ajwfrost** [comment]
End of June should have been the .98 version (very good idea to add the dates into the release notes!) as that was early April..
I will pass on a few comments to the folk about the website release notes and the invalid-download file, generally you can get around that by refreshing the browser page and trying again. If you've got the SDK on your computer then it should match up in terms of the root folder with the version information in the PDF as well as the contents of the air-sdk-description.xml file.

FYI we're also going to be adding links to previous releases, at least from the .98 build onwards..

thanks

**Comment by ventr1x** [comment]
The problem seems to be completely gone with .98. 
Thanks for you help.

A little side question: how long is your average response time on mails? And is there perhaps a phone number for more urgent problems (this was the first time needed, so we're not too annoying ;))?

P.S.: your license renewal emails include localhost urls, so it's impossible to renew anything right now.

**Comment by ajwfrost** [comment]
Hi - the email response time varies more than I'd like .. that inbox is somewhat busy as it receives a number of notifications as well as customer emails, we're trying to get some better processes to be able to filter the 'real' customer concerns. Generally there are around 3 or 4 folk who monitor it periodically and then if necessary they will raise things in our meetings - but with this github site, Daniel adjusted the configuration so that I get notified of every new issue to my personal mailbox. Generally I then forward an issue to the appropriate team member :-)

I saw the localhost thing, I assume the guys are aware of that and working on it - will check though!

thanks
